app/models/operatingsystems/debian.rb

```diff
@@ -16,7 +16,7 @@ def preseed_server(medium_provider)
   end
 
   def preseed_path(medium_provider)
-    medium_provider.medium_uri.select(:path, :query).compact.join('?')
+    medium_provider.medium_uri(&method(:transform_vars)).select(:path, :query).compact.join('?')
```

Comment:
This would mean, that someone enters a EL path, like arch=x86_64. Does this make sense? Is this something we need to support?

Reply:
Do you mean that `$arch` would be interpolated as `aarch64`?  That's what currently happens (or was happening, when I was looking at it) without this patch, the EL interpolation, for the iso path.

My description in the [issue](https://projects.theforeman.org/issues/36830) may be clarifying (if I'm interpreting what you're saying correctly).  The impetus for this patch was that I used `/pub/installation_media/ubuntu/$version-$arch/` in the Path of my Installation Medium, and *got* `aarch64` when I was expecting `arm64` (and would similarly, I believe, get `x86_64` instead of `amd64`).

It's not a *need* strictly, but I think the current behavior violates the principle of least astonishment, that I would need to rename my Ubuntu media from `arm64` *to* `aarch64` in order to create a generalized Installation Medium using the tokens the help text offers.  The behavior is also different (!) than what happens with `boot_file_sources`, so you end up having to do
```
22.04.3-arm64/
22.04.3-aarch64.iso
```
i.e. name the directory `22.04.3-arm64` but the iso `22.04.3-aarch64.iso`, if you don't use this patch.

Reply:
Ah. Now I understand your use-case. Then I think this make's sense. Thanks a lot for your clarification. 

Regarding the test @ekohl asked for, you may have a look at the following test, which may can be copied and then adapted to this use-case:

https://github.com/theforeman/foreman/blob/develop/test/unit/foreman/renderer/scope/variables/base_test.rb#L27
which uses https://github.com/theforeman/foreman/blob/develop/test/factories/host_related.rb#L247 and therefore https://github.com/theforeman/foreman/blob/develop/test/factories/operatingsystem.rb#L111.

Maybe it should use something similar than https://github.com/theforeman/foreman/blob/develop/test/factories/operatingsystem.rb#L156 with a medium similar than https://github.com/theforeman/foreman/blob/develop/test/factories/medium.rb#L19 but with a "arch" url.